packages/core/src/asciidoctor-core-api.js

```diff
@@ -3047,3 +3047,402 @@ Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
 SyntaxHighlighterBase.prototype.registerFor = function (names) {
   SyntaxHighlighter['$register'](this, names)
 }
+
+// Table API
+
+/**
+ * Methods for managing AsciiDoc tables.
+ * @namespace
+ * @extends AbstractBlock
+ */
+var Table = Opal.Asciidoctor.Table
+
+/**
+ * Create a new Table element
+ * @param {AbstractBlock} parent
+ * @param {Object|undefined} attributes
+ * @returns {Table}
+ */
+Table.create = function (parent, attributes) {
+  return this.$new(parent, attributes)
+}
+
+/**
+ * Check, if a given option is set
+ * @param {string} opt the option to check
+ * @returns {boolean}
+ */
+Table.prototype.hasOption = function (opt) {
```

Comment:
As a first step I would like to map only existing methods from Asciidoctor core and then we can add utility functions like this one.
Could you please move utility functions in another pull request?

Reply:
Done. Will be included in the next commit